src/App.js

```diff
@@ -1,50 +1,69 @@
-import  React, {useEffect} from 'react';
+import  React, { useEffect, useState } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
-import {useState} from 'react';
 
 
-function useSemiPersistentState () {
- const [todoList, setTodoList]=
-   useState((JSON.parse(localStorage.getItem('savedTodoList')) || []));
-  
-
-
-  useEffect(() => {
-   localStorage.setItem('savedTodoList', JSON.stringify(todoList));
- }, [todoList]);
+        function App() {
+               
+        const [todoList, setTodoList]= useState (
+       
+        JSON.parse(localStorage.getItem('savedTodoList')) || []
 
+        );
 
- return[todoList, setTodoList];
+        const [isLoading, setIsLoading] = React.useState(true);
+  
 
+   useEffect(() => {
+    const loadTodoList = new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve({ data: { todoList: todoList  } });
+    }, 2000);
+  
+  });
 
-};
-function App() {
- const[todoList, setTodoList] = useSemiPersistentState();
+  loadTodoList
+      .then((result) => {
+        
+        setTodoList(result.data.todoList);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.error(error);
+        setIsLoading(false);
+      });
+  }, [todoList]);
 
+  function addTodo(newTodo) {
+    setTodoList([...todoList, newTodo]);
+  }
 
- function addTodo(newTodo) {
-   setTodoList([...todoList, newTodo]);
-   }
-  function removeTodo(id){
-   const newTodoList = todoList.filter(todo => todo.id !== id);
-   setTodoList(newTodoList);
-  
+  function removeTodo(id) {
+    const newTodoList = todoList.filter((todo) => todo.id !== id);
+    setTodoList(newTodoList);
   }
- return (
 
+ return (
+<>
 
   <div>
     <header>
     <h1>Todo List</h1>
       <h3>Trendyfloristapp</h3>
       </header>
-     
+      
       <TodoList todoList={todoList} onRemoveTodo={removeTodo} />
-       <AddTodoForm onAddTodo={addTodo}/>    
+      {isLoading ? (
+        <p>Loading...</p>
+      ) : (
+     
+        
+        <AddTodoForm onAddTodo={addTodo}/>
+      )}
```

Comment:
It would be best if you load todoList 
<AddTodoForm onAddTodo={addTodo} />

       {isLoading ? (
         <p>Loading...</p>
       ) : (
         <TodoList todoList={todoList} onRemoveTodo={removeTodo} />
       )}

Reply:
Hi 
XeniyaDob, 
 Thanks a lot for your time to review my work.  I appreciate your help and suggestions.
  I will try to improve the app as you suggested.